main/simulator.cpp

```diff
@@ -1 +1,81 @@
 #include "main/simulator.h"
+
+std::vector<std::unique_ptr<objects::EnergyCollector>>
+buildCollectors(const ModelInterface &model, int numCollectors) {
+
+  if (model.empty()) {
+    throw std::invalid_argument("Empty model");
+  }
+  if (numCollectors < 4) {
+    std::stringstream ss;
+    ss << "numCollectors: " << numCollectors << " is less then 4";
+    throw std::invalid_argument(ss.str());
+  }
+
+  // We distribute collectors along 4 quarter circles above the model, so
+  // |numCollectors| has to be divisible by 4. Or |numCollectors|-1 has to be
+  // divisible by 4 (if it is odd), in which case there is one more collector
+  // placed right above the model.
+  if (numCollectors % 4 != 0 && (numCollectors - 1) % 4 != 0) {
+    std::stringstream ss;
+    ss << "numCollectors or numCollectors-1 has to be divisible by 4, got "
+       << "numCollectors = " << numCollectors;
+    throw std::invalid_argument(ss.str());
+  }
+
+  // We are placing collectors on two half-circles, such that distance between
+  // each collector on a given circle is equal to their radius.
+
+  // For even number of collectors, on each half-circle we place N/2 collectors
+  // in total, with 2 collectors at the ground level. This divides the PI into
+  // N/2-1 segments.
+  // For odd number of collectors, we do the same and place 1 more collector at
+  // the top, so we divide PI into (N - 1) / 2 segments.
+  // Those two cases can be shorten into:
+  const int numCollectorReminder = numCollectors % 2;
+  const float angleBetweenCollectors =
+      2 * constants::kPi / (numCollectors + numCollectorReminder - 2.0f);
```

Comment:
Why 2.0f? Do you really have to use floats here? I had an impression that float / int -> float. Is it not?